### PR TITLE
Added NBTTagLongArray

### PR DESCRIPTION
**Pull request by Tobi14601**
Implemented NBTTagLongArray

### CODE REVIEW

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		int j = datainput.readInt();
+		if (j >= 1 << 24) throw new NBTLoadException("Error while loading LongArray");
+
+		nbtreadlimiter.allocate((long) (64 * j));
+		this.data = new long[j];
+
+		for (int k = 0; k < j; ++k) {
+			this.data[k] = datainput.readLong();
+		}
+
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:42]: remove this whitespace

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	
+	@Override
+	public String toString() {
+		String s = "[";
+		long[] along = this.data;
+		int i = along.length;
+
+		for (int j = 0; j < i; ++j) {
+			long k = along[j];
+
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:76]: remove this whitespace

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+			s = s + k + ",";
+		}
+
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:79]: remove this whitespace

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	}
+
+	@Override
+	void write(DataOutput dataoutput) throws IOException {
+		dataoutput.writeInt(this.data.length);
+
+		for (int i = 0; i < this.data.length; ++i) {
+			dataoutput.writeLong(this.data[i]);
+		}
+
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:28]: remove this whitespace

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	
+	@Override
+	public boolean equals(Object object) {
+		return super.equals(object) && Arrays.equals(this.data, ((NBTTagLongArray) object).data);
+	}
+	
+	@Override
+	public String toString() {
+		String s = "[";
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:70]: String prefix

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+		long[] along = this.data;
+		int i = along.length;
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:72]: int length

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+		for (int j = 0; j < i; ++j) {
+			long k = along[j];
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:75]: some other naming? alongElement?

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+	public NBTBase clone() {
+		long[] along = new long[this.data.length];
+
+		System.arraycopy(this.data, 0, along, 0, this.data.length);
+		return new NBTTagLongArray(along);
+	}
+	
+	@Override
+	public boolean equals(Object object) {
+		return super.equals(object) && Arrays.equals(this.data, ((NBTTagLongArray) object).data);
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:65]: check if object is null

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+		for (int i = 0; i < this.data.length; ++i) {
+			dataoutput.writeLong(this.data[i]);
+		}
+
+	}
+
+	@Override
+	void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
+		int j = datainput.readInt();
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:33]: some other name? what exactly is "j"?

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+		dataoutput.writeInt(this.data.length);
+
+		for (int i = 0; i < this.data.length; ++i) {
+			dataoutput.writeLong(this.data[i]);
+		}
+
+	}
+
+	@Override
+	void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:32]: some other name, what exactly is "i"?
datainput -> dataInput
nbtreadlimiter -> nbtReadLimiter

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+	NBTTagLongArray() {
+	}
+
+	public NBTTagLongArray(long[] along) {
+		this.data = along;
+	}
+
+	@Override
+	void write(DataOutput dataoutput) throws IOException {
```
Comment (Kroseida) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:22]: dataOutput

**Review by Kroseida — CHANGES_REQUESTED** [review]

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.Arrays;
+
+public class NBTTagLongArray extends NBTBase {
+
+	private long[] data;
+
+	NBTTagLongArray() {
+	}
+
+	public NBTTagLongArray(long[] along) {
```
Comment (ItsKev) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:17]: Rename to data or something similar.

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+		for (int i = 0; i < this.data.length; ++i) {
+			dataoutput.writeLong(this.data[i]);
+		}
+
+	}
+
+	@Override
+	void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
+		int j = datainput.readInt();
+		if (j >= 1 << 24) throw new NBTLoadException("Error while loading LongArray");
```
Comment (ItsKev) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:34]: Use curly braces around the body

File: src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	
+	@Override
+	public long[] getData() {
+		return this.data;
+	}
+
+	@Override
+	public NBTBase clone() {
+		long[] along = new long[this.data.length];
```
Comment (ItsKev) [inline src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java:57]: Rename this to something else. A long variable shouldn't include its type in the name itself

**Review by ItsKev — CHANGES_REQUESTED** [review]

**Review by hbrombeer — APPROVED** [review]
sieht gut aus

**Review by Kroseida — APPROVED** [review]

**Review by ItsKev — APPROVED** [review]

**Review by lusu007 — APPROVED** [review]

**Pull request merged**